packages/react/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,158 @@
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import * as React from 'react'
+import * as _ from 'lodash'
+import {
+  UIComponentProps,
+  UIComponent,
+  createShorthandFactory,
+  ShorthandFactory,
+  commonPropTypes,
+  applyAccessibilityKeyHandlers,
+  childrenExist,
+  ChildrenComponentProps,
+} from '../../lib'
+import { WithAsProp, withSafeTypeForAs, ShorthandCollection, ShorthandValue } from '../../types'
+import Box, { BoxProps } from '../Box/Box'
+import Button, { ButtonProps } from '../Button/Button'
+
+export interface CarouselProps extends UIComponentProps, ChildrenComponentProps {
```

Comment:
Are we going to have `autoPlay` prop and `picker` for selecting specific item? I understand this is a draft PR, not sure what will be final solution look like